src/test/java/mate/academy/service/RoleServiceTest.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import mate.academy.service.impl.RoleServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class RoleServiceTest {
+    private static final String ROLE_NAME = "USER";
+    private static final String WRONG_ROLE_NAME = "HUMAN";
+    private RoleService roleService;
+    private RoleDao roleDao;
+    private Role role;
+
+    @BeforeEach
+    void setUp() {
+        roleDao = Mockito.mock(RoleDao.class);
+        roleService = new RoleServiceImpl(roleDao);
+        role = new Role(Role.RoleName.USER);
+    }
+
+    @Test
+    void save_ok() {
+        Mockito.when(roleDao.save(role)).thenReturn(role);
+        Role actual = roleDao.save(role);
```

Comment:
Use roleService.save(role) instead of roleDao.save(role) to test the service layer.